cd/auto-update/auto-update-controllers.sh

```diff
@@ -0,0 +1,183 @@
+pushd "$WORKSPACE_DIR" >/dev/null
+  CONTROLLER_NAMES=$(find . -maxdepth 1 -name "*-controller" -type d | cut -d"/" -f2)
+popd >/dev/null
+
+for CONTROLLER_NAME in $CONTROLLER_NAMES; do
+  SERVICE_NAME=$(echo "$CONTROLLER_NAME"| sed 's/-controller$//g')
+  CONTROLLER_DIR="$WORKSPACE_DIR/$CONTROLLER_NAME"
+
+  echo -n "auto-update-controllers.sh][INFO] Ensuring that GitHub label $GITHUB_LABEL exists for $GITHUB_ORG/$CONTROLLER_NAME ... "
+  if ! gh api repos/"$GITHUB_ORG"/"$CONTROLLER_NAME"/labels/"$GITHUB_LABEL" --silent >/dev/null; then
+    echo ""
+    echo "auto-update-controllers.sh][INFO] Could not find label $GITHUB_LABEL in repo $GITHUB_ORG/$CONTROLLER_NAME"
+    echo -n "Creating new GitHub label $GITHUB_LABEL ... "
+    if ! gh api -X POST repos/"$GITHUB_ORG"/"$CONTROLLER_NAME"/labels -f name="$GITHUB_LABEL" -f color="$GITHUB_LABEL_COLOR" >/dev/null; then
+      echo ""
+      echo "auto-update-controllers.sh][ERROR] Failed to create label $GITHUB_LABEL. Skipping $CONTROLLER_NAME"
+      continue
+    else
+      echo "ok"
+    fi
+  else
+    echo "ok"
+  fi
+
+  echo "auto-update-controllers.sh][INFO] Updating project description files of existing controller using command 'make run'"
+  export SERVICE=$SERVICE_NAME
+  MAKE_RUN_OUTPUT_FILE=/tmp/"$SERVICE_NAME"_make_run_output
+  MAKE_RUN_ERROR_FILE=/tmp/"$SERVICE_NAME"_make_run_error
+  if ! make run > "$MAKE_RUN_OUTPUT_FILE" 2> "$MAKE_RUN_ERROR_FILE"; then
+    cat "$MAKE_RUN_ERROR_FILE"
+
+    echo "auto-update-controllers.sh][ERROR] Failure while executing 'make run' command. Creating/Updating GitHub issue"
+    ISSUE_TITLE="Errors while updating project description files for $CONTROLLER_NAME"
+
+    # Capture 'make run' command output & error, then persist
+    # in '$GITHUB_ISSUE_BODY_FILE'
+    MAKE_RUN_OUTPUT=$(cat "$MAKE_RUN_OUTPUT_FILE")
+    MAKE_RUN_ERROR_OUTPUT=$(cat "$MAKE_RUN_ERROR_FILE")
+    GITHUB_ISSUE_BODY_TEMPLATE_FILE="$THIS_DIR/gh_issue_update_controller_template.txt"
+    GITHUB_ISSUE_BODY_FILE=/tmp/"$SERVICE_NAME"_gh_issue_update_controller
+    eval "echo \"$(cat "$GITHUB_ISSUE_BODY_TEMPLATE_FILE")\"" > $GITHUB_ISSUE_BODY_FILE
+
+    open_gh_issue "$GITHUB_ISSUE_ORG_REPO" "$ISSUE_TITLE" "$GITHUB_ISSUE_BODY_FILE"
+    # Skip creating PR for this service controller after updating GitHub issue.
+    continue
+  fi
+
+  # Since there are no failures, print make run output in prowjob logs
+  cat "$MAKE_RUN_OUTPUT_FILE"
+  pushd "$CONTROLLER_DIR" >/dev/null
+    GITHUB_CONTROLLER_ORG_REPO="$GITHUB_ORG/$CONTROLLER_NAME"
+
+    # add git remote
+    echo -n "auto-update-controllers.sh][INFO] Adding git remote ... "
+    if ! git remote add origin "https://github.com/$GITHUB_ORG/$CONTROLLER_NAME.git" >/dev/null; then
+      echo ""
+      echo "auto-update-controllers.sh][ERROR] Unable to add git remote. Skipping $CONTROLLER_NAME"
+      continue
+    fi
+    echo "ok"
+
+    # Add all the files & create a GitHub commit
+    git add .
```

Comment:
I think before this you can add a check to make sure there are files in the git diff. 

Something like
```
fc=$(git diff --name-only | cat | wc -l | tr -d ' ')
if [[ $fc -eq 0 ]]; then
    echo "auto-update-controllers.sh][ERROR] no changes to commit"
    continue
fi
```